File: java-junit5/src/test/java/org/codecop/Session2d_WordCounterTest.java
package org.codecop;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Arrays;

import org.junit.jupiter.api.Test;

/**
 * Bonus Session 2d: WordCounterTest - Assertions with AssertJ. <br />
 * @see "https://assertj.github.io/doc/#assertj-core-quick-start"
 *
 * @author Daniel Passecker
 */
class Session2d_WordCounterTest {

    // TODO Add the proper assertions to complete the tests using AssertJ.
    // The test name explains what needs to be asserted.

    @Test
    void shouldCountNumberOfWords() {
        WordCounter counter = new WordCounter("Keep the bar green to keep the code clean.");
        assertEquals(9, counter.numberOfWords());
        // AssertJ is fluent:
        // TODO Check that counter.numberOfWords(), isEqualTo 9.
    }

    @Test
    void shouldContainUniqueWord() {
        WordCounter counter = new WordCounter("green bar green hat");
        assertTrue(Arrays.asList(counter.uniqueWords()).contains("bar"));
        // AssertJ is fluent:
        // TODO Check that counter.uniqueWords() contains "bar".
    }

    @Test
    void shouldNotContainUniqueWord() {
        WordCounter counter = new WordCounter("green bar green hat");
        assertFalse(Arrays.asList(counter.uniqueWords()).contains("foo"));
        // AssertJ is fluent:
        // TODO Check that counter.uniqueWords() doesNotContain "foo".
    }

    @Test
    void shouldFindNumberOfUniqueWords() {
        WordCounter counter = new WordCounter("green bar green hat");
        assertEquals(3, counter.uniqueWords().length);
        // AssertJ is fluent:
        // TODO Check that counter.uniqueWords() hasSize 3.
    }
}

File: csharp-xunit/WordCount.Tests/Session2d_WordCounterTest.cs
using FluentAssertions;
using Xunit;

namespace Org.Codecop.WordCount.Tests
{
    /// <summary>Bonus Session 2d: WordCounterTest - Assertions with FluentAssertions.</summary>
    /// <seealso>"https://fluentassertions.com/documentation/"</seealso>
    public class Session2d_WordCounterTest
    {

        // TODO Add the proper assertions to complete the tests using AssertJ.
        // The test name explains what needs to be asserted.

        [Fact]
        public void ShouldCountNumberOfWords()
        {
            var counter = new WordCounter("Keep the bar green to keep the code clean.");
            Assert.Equal(9, counter.NumberOfWords());
            // FluentAssertions is fluent:
            // TODO Check that counter.NumberOfWords() and 9 are equal.
        }

        [Fact]
        public void ShouldContainUniqueWord()
        {
            var counter = new WordCounter("green bar green hat");
            Assert.Contains(counter.UniqueWords(), w => w == "bar");
            // FluentAssertions is fluent:
            // TODO Check that counter.UniqueWords() does contain "bar".
        }

        [Fact]
        public void ShouldNotContainUniqueWord()
        {
            var counter = new WordCounter("green bar green hat");
            Assert.DoesNotContain(counter.UniqueWords(), w => w == "foo");
            // FluentAssertions is fluent:
            // TODO Check that counter.UniqueWords() does not contain "foo".
        }

        [Fact]
        public void ShouldFindNumberOfUniqueWords()
        {
            var counter = new WordCounter("green bar green hat");
            Assert.Equal(3, counter.UniqueWords().Length);
            // FluentAssertions is fluent:
            // TODO Check that counter.UniqueWords() does have count of 3.
        }
    }
}
